deps/chakrashim/src/jsrtutils.cc

```diff
@@ -977,33 +977,46 @@ StringUtf8::~StringUtf8() {
 }
 
 JsErrorCode StringUtf8::LengthFrom(JsValueRef strRef) {
-    CHAKRA_ASSERT(_length == 0);
+  CHAKRA_ASSERT(_length == 0);
 
-    size_t len = 0;
-    IfJsErrorRet(JsCopyString(strRef, nullptr, 0, nullptr, &len));
+  size_t len = 0;
+  IfJsErrorRet(JsCopyString(strRef, nullptr, 0, nullptr, &len));
 
-    _length = len;
-    return JsNoError;
+  _length = len;
+  return JsNoError;
 }
 
 JsErrorCode StringUtf8::From(JsValueRef strRef) {
   CHAKRA_ASSERT(!_str);
 
-  IfJsErrorRet(LengthFrom(strRef));
-  size_t len = length();
+  int strLength = 0;
+  IfJsErrorRet(JsGetStringLength(strRef, &strLength));
 
-  char* buffer = reinterpret_cast<char*>(malloc(len+1));
-  CHAKRA_VERIFY(buffer != nullptr);
+  // assume string contains ascii characters only
+  _str = reinterpret_cast<char*>(malloc(strLength + 1));
 
   size_t written = 0;
-  IfJsErrorRet(JsCopyString(strRef, buffer, len, &written, nullptr));
+  size_t actualLength = 0;
+  IfJsErrorRet(JsCopyString(strRef, _str, strLength, &written, &actualLength));
+  CHAKRA_ASSERT(strLength == written);
```

Comment:
This isn't guaranteed to be true, if the buffer stops mid-multibyte character. E.g. if there is a 2 byte Utf8 character which would start at position `strLength-1`, then `JsCopyString` will omit all bytes for that codepoint from the buffer, I believe, so `written` will be `strLength-1`. Similarly if it ends at a 3 byte character, it can be `strLength-2`.